docker/nginx-consul-template/nginx.conf.tmpl

```diff
@@ -0,0 +1,24 @@
+events {
+    worker_connections 4096;
+}
+
+http {
+    upstream service {
+      {{- range service (printf "%s.%s" (env "DE_ENV") (env "NGINX_PROXY_SERVICE_NAME")) }}
+      server {{.Address}}:{{.Port}} max_fails=3 fail_timeout=60 weight=1 {{- if not (.Tags.Contains (key (printf "%s/%s/color" (env "DE_ENV") (env "NGINX_PROXY_SERVICE_NAME"))))}} backup{{end}}; # {{.ID}} {{.Tags}}
+      {{- else}}
+      server 127.0.0.1:65535; # Force a 502. No {{env "DE_ENV"}}.{{env "NGINX_PROXY_SERVICE_NAME"}} services were found.
```

Comment:
I've changed this to completely replace the contents of the `http` block when no servers can be found. In the case where it can, it puts in the upstream etc. as usual, but where it can't, it just puts a simple server block with `return 502` directly. This returns an HTML default nginx 502 error, which we might want to replace instead with some sort of standard JSON response (which should be possible with an `error_page` declaration, I believe). Thoughts?